airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestination.java

```diff
@@ -0,0 +1,184 @@
+  private static final String AUTH_TYPE = "auth_type";
+  private static final String USERNAME = "username";
+  private static final String PASSWORD = "password";
+  private static final String AUTHORIZATION = "authorization";
+  private static final String LOGIN_AND_PASSWORD = "login/password";
+  private static final String AIRBYTE_DATA = "_airbyte_data";
+
+  private final MongoDBNameTransformer namingResolver;
+
+  public MongodbDestination() {
+    namingResolver = new MongoDBNameTransformer();
+  }
+
+  public static void main(String[] args) throws Exception {
+    final Destination destination = new MongodbDestination();
+    LOGGER.info("starting destination: {}", MongodbDestination.class);
+    new IntegrationRunner(destination).run(args);
+    LOGGER.info("completed destination: {}", MongodbDestination.class);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      var uri = getMongoClientURI(config);
+      var client = getMongoClient(config.get(HOST).asText(), config.get(PORT).asInt(), uri);
+      var database = getMongoDatabase(client, config.get(DATABASE).asText());
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
+    } catch (RuntimeException e) {
+      LOGGER.info("Check failed.", e);
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.FAILED)
+          .withMessage(e.getMessage() != null ? e.getMessage() : e.toString());
+    }
+  }
+
+  @Override
+  public AirbyteMessageConsumer getConsumer(JsonNode config,
+                                            ConfiguredAirbyteCatalog catalog,
+                                            Consumer<AirbyteMessage> outputRecordCollector) {
+    var uri = getMongoClientURI(config);
+    var client = getMongoClient(config.get(HOST).asText(), config.get(PORT).asInt(), uri);
+    var database = getMongoDatabase(client, config.get(DATABASE).asText());
+
+    Map<AirbyteStreamNameNamespacePair, MongodbWriteConfig> writeConfigs = new HashMap<>();
+    for (final ConfiguredAirbyteStream configStream : catalog.getStreams()) {
+      final AirbyteStream stream = configStream.getStream();
+      final String streamName = stream.getName();
+      final String collectionName = namingResolver.getRawTableName(streamName);
+      final String tmpCollectionName = namingResolver.getTmpTableName(streamName);
+
+      if (DestinationSyncMode.OVERWRITE == configStream.getDestinationSyncMode()) {
+        database.getCollection(collectionName).drop();
+      }
+
+      MongoCollection<Document> collection = getOrCreateNewMongodbCollection(database, tmpCollectionName);
+      Set<Document> documents = new HashSet<>();
+      try (MongoCursor<Document> cursor = collection.find().projection(excludeId()).iterator()) {
+        while (cursor.hasNext()) {
+          documents.add(cursor.next().get(AIRBYTE_DATA, Document.class));
+        }
+      }
```

Comment:
All documents within a collection is read and stored in the `documents` variable. Will this cause out-of-memory issue if the collection is huge?

Reply:
Great moment. Yes, with a large number of records, we will get memory problems. Thank you for pointing this out. I added a hashCode field for mongodb to the document. Now documents will be compared by this field. This should fix the problem you pointed out.